mlflow/sklearn.py

```diff
@@ -424,3 +424,29 @@ def load_model(model_uri):
     return _load_model_from_local_file(
         path=sklearn_model_artifacts_path, serialization_format=serialization_format
     )
+
+
+class _SklearnTrainingSession(object):
```

Comment:
Can we include some information about the lack of thread safety guarantees for this class and why that's okay for now? I.e., 1. we don't currently have any use cases for `allow_children=True` and 2. the list `append` & `pop` operations are thread-safe, so we will always clear the session stack once all `SklearnTrainingSessions` exit.

In the future, we'll need to figure out how to handle multithreaded backends.